### PR TITLE
Add `stopAt` option

### LINKED ISSUE #49: Add `stopAt` option
This would let you stop the search before reaching root if there were no matches before the `stopAt` directory.

Suggestions for a better option name welcome:
- `stopAtDirectory`
- ?

Previous discussion: https://github.com/sindresorhus/find-up/issues/27

### PR DESCRIPTION
**Pull request by vast**
Fixes #49.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -100,6 +100,13 @@ Default: `true`
 
 Allow symbolic links to match if they point to the chosen path type.
 
+##### stopAt
+
+Type: `string`\
+Default: `path.parse(cwd).root`
+
+The absolute path to the directory to stop the search before reaching root if there were no matches before the `stopAt` directory.
```
Comment (sindresorhus) [inline readme.md:108]: Why does it have to be an *absolute* path? I think you can just `path.resolve` the input path you get from the user.

**Comment by sindresorhus** [comment]
Thanks for working on this. You need to add types to index.d.ts too.

**Comment by vast** [comment]
Thanks for the feedback. Fixed both issues in:
* https://github.com/vast/find-up/commit/7fda6df47d95042d04c7ae191b689237ce9844a5
* https://github.com/vast/find-up/commit/2bd5f2e6b944e76a1c20d887daeff85cc9b2de8c

Could you please have a look?

File: index.d.ts [hunk 2]
```diff
@@ -8,6 +8,8 @@ export const findUpStop: unique symbol;
 
 export type Match = string | typeof findUpStop | undefined;
 
+export type Options = LocatePathOptions & {stopAt?: string};
```
Comment (sindresorhus) [inline index.d.ts:11]: The `stopAt` option needs docs: https://github.com/sindresorhus/typescript-definition-style-guide

File: test.js [hunk 3]
```diff
@@ -210,6 +211,24 @@ test('sync (cousin file, custom cwd)', t => {
...
 });
 
+test('async (cousin file, custom cwd with stopAt)', async t => {
+	const foundPath = await findUp(name.baz, {
+		cwd: relative.barDir,
+		stopAt: absolute.fooDir,
+	});
+
+	t.is(foundPath, undefined);
+});
```
Comment (sindresorhus) [inline test.js:221]: Can you also do a test where the foundPath is the `stopAt` path?

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 2 above)
Comment (vast) [reply index.d.ts]: Added docs:
https://github.com/sindresorhus/find-up/pull/54/commits/c2134ed942e32804dd16c07a28e40d5e4809c521

File: test.js (hunk 3 above)
Comment (vast) [reply test.js:221]: Yep, added:
https://github.com/sindresorhus/find-up/pull/54/commits/d2a1b0d6f7fbc6371314b096be4fdf82ddff0956

**Pull request merged**